validation.py

```diff
@@ -0,0 +1,492 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import glob
+import argparse
+import random
+import functools
+import itertools
+import multiprocessing
+import time
+
+import numpy as np
+import tskit
+# Work around an issue on systems with large numbers of cores.
+# https://github.com/cggh/scikit-allel/issues/285
+os.environ["NUMEXPR_MAX_THREADS"] = f"{os.cpu_count()}"  # NOQA
+import allel
+
+import matplotlib
+matplotlib.use('Agg')  # NOQA # don't try to use $DISPLAY
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+
+import stdpopsim
+import stdpopsim.cli
+
+
+def warning(msg):
+    """
+    Print a warning, with output less ugly than that of warnings.warn().
+    """
+    print(f"WARNING: {msg}", file=sys.stderr)
+
+
+def do_cmd(cmd, out_dir, seed):
+    cmd = cmd.split()
+    assert "-o" not in cmd and "--output" not in cmd
+    assert "-s" not in cmd and "--seed" not in cmd
+    out_file = os.path.join(out_dir, f"{seed}.trees")
+    full_cmd = cmd + f" -q -o {out_file} -s {seed}".split()
+    t0 = time.time()
+    stdpopsim.cli.stdpopsim_main(full_cmd)
+    t1 = time.time()
+    assert os.path.exists(out_file)
+    return out_file, t1 - t0
+
+
+"""
```

Comment:
Slightly odd to use a docstring here, and it might have unintended consequences (I'm not sure what this docstring gets attached to). Best use a comment (banner lines are fine IMO, if you want to split up the file visually)